File: lib/to_mdx/parsers/open_document/office/body.rb
# frozen_string_literal: true

require_relative "../base"

module ToMdx
  module Parsers
    module Office
      class Body < Base
        def to_mdx
          contentful_children.map(&:to_mdx).join
        end

        def to_s
          contentful_children.map(&:to_s).join
        end

        register_for "office:body"
      end
    end
  end
end

File: src/parsers/open-document/office/body.ts
import type { ParseContext } from "../../base_element.ts";
import { BaseElement } from "../../base_element.ts";

export class Body extends BaseElement {
  constructor(element: Element, context: ParseContext, parent: BaseElement | null) {
    super(element, context, parent);
  }

  toMdx(): string {
    return this.contentfulChildren()
      .map((child) => child.toMdx())
      .join("");
  }

  toString(): string {
    return this.contentfulChildren()
      .map((child) => child.toString())
      .join("");
  }

  static {
    BaseElement.registerFor.call(this, "office:body");
  }
}
